src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -218,6 +218,16 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     /// FIXME Why do we need this flag? What's difference from Temporary and DeleteOnDestroy state? Can we get rid of this?
     bool is_temp = false;
 
+    enum class BlobsRemovalPolicyForTemporaryParts
+    {
+        ASK_KEEPER,
+        REMOVE_BLOBS,
+        PRESERVE_BLOBS,
```

Comment:
It would be great to add a comment that explains why we need different behavior here, like:
 - ASK_KEEPER - unlock shared data in ZooKeeper, as usual
 - REMOVE_BLOBS - sometimes we know for sure that we have just written new blobs and other replicas have not seen them yet, so we can safely remove these blobs if we fail to commit a new part (and there's nothing to unlock in ZooKeeper if the part was temporary). It avoids leaving some garbage on S3
 - PRESERVE_BLOBS - ...